fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt

```diff
@@ -0,0 +1,138 @@
+import com.microsoft.fluentui.theme.token.ControlTokens
+import com.microsoft.fluentui.theme.token.Icon
+import com.microsoft.fluentui.theme.token.controlTokens.ProgressTextInfo
+import com.microsoft.fluentui.theme.token.controlTokens.ProgressTextTokens
+import com.microsoft.fluentui.util.dpToPx
+
+/**
+ * ProgressTexts are used to give information through a text about the current progress.
+ * A ProgressText consists of a Text and a progressbar.
+ * @param text Text or info to display
+ * @param progress Progress of the progress indicator. 0.0 represents no progress and 1.0 represents full progress.
+ * @param onCancelClick onClick method to execute when cancel button is clicked
+ * @param modifier Modifier for the progress text
+ * @param progressTextTokens Token values for the ProgressText
+ */
+@Composable
+fun ProgressText(
+    text: String,
+    progress: Float,
+    onCancelClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    progressTextTokens: ProgressTextTokens? = null
+) {
+    val tokens = progressTextTokens
+        ?: FluentTheme.controlTokens.tokens[ControlTokens.ControlType.ProgressText] as ProgressTextTokens
+    val progressTextInfo = ProgressTextInfo()
+    val currentProgress = animateFloatAsState(
+        targetValue = progress.coerceIn(0f..1f), animationSpec = tween(
+            delayMillis = 0, durationMillis = 1000, easing = LinearOutSlowInEasing
+        )
+    )
+    val backgroundBrush = tokens.backgroundBrush(progressTextInfo = progressTextInfo)
+    val border = tokens.borderWidth(progressTextInfo = progressTextInfo)
+    val borderColor = tokens.borderColor(progressTextInfo = progressTextInfo)
+    val typography = tokens.typography(progressTextInfo = progressTextInfo)
+    val textColor = tokens.textColor(progressTextInfo = progressTextInfo)
+    val progressIndicatorHeight = tokens.progressbarHeight(progressTextInfo = progressTextInfo)
+    val progressIndicatorBackgroundColor =
+        tokens.progressbarBackgroundColor(progressTextInfo = progressTextInfo)
+    val progressIndicatorColor = tokens.progressbarBrush(progressTextInfo = progressTextInfo)
+    val iconTextSpacing = tokens.iconTextSpacing(progressTextInfo = progressTextInfo)
+    val padding = tokens.padding(progressTextInfo = progressTextInfo)
+    val iconSize = tokens.iconSize(progressTextInfo = progressTextInfo)
+    val iconColor = tokens.iconColor(progressTextInfo = progressTextInfo)
+    val shape = RoundedCornerShape(12.dp)
+    Box(modifier = Modifier.clip(shape)) {
+        Box(
+            modifier = modifier
+                .background(
+                    backgroundBrush, shape
+                )
+                .border(
+                    border, borderColor, shape
+                )
+        ) {
+            Row(
+                modifier = Modifier.padding(padding),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Icon(
+                    ProgressTextIcons.DismissCircle,
```

Comment:
I think the icon should be changeable as per partner requirement.

Reply:
Removed oncancelclick and added parameter to take icon as a fluenticon. Onclick can be given through fluent icon